baker/redirects.ts

```diff
@@ -84,3 +88,89 @@ export const getRedirects = async () => {
 
     return redirects
 }
+
+export const getGrapherAndWordpressRedirectsMap = memoize(
+    async (): Promise<Map<string, string>> => {
+        // source: pathnames only (e.g. /transport)
+        // target: pathnames with or without origins (e.g. /transport-new or https://ourworldindata.org/transport-new)
+        const redirects = new Map()
+
+        // todo(refactor): export as function to reuse in getRedirects?
+        const chartRedirectRows = await db.queryMysql(`
+        SELECT chart_slug_redirects.slug, JSON_EXTRACT(charts.config, "$.slug") as trueSlug
+        FROM chart_slug_redirects INNER JOIN charts ON charts.id=chart_id
+    `)
+
+        // todo(refactor) : export as function to reuse in getRedirects?
+        const wordpressRedirectRows = await wpdb.singleton.query(
+            `SELECT url, action_data FROM wp_redirection_items WHERE status = 'enabled'`
+        )
+
+        // The order the redirects are added to the map is important. Adding the
+        // Wordpress redirects last means that Wordpress redirects can overwrite
+        // grapher redirects.
+        for (const row of chartRedirectRows) {
+            const trueSlug = JSON.parse(row.trueSlug)
+            if (row.slug !== trueSlug) {
+                redirects.set(`/grapher/${row.slug}`, `/grapher/${trueSlug}`)
+            }
+        }
+
+        for (const row of wordpressRedirectRows) {
+            redirects.set(row.url, row.action_data)
+        }
+
+        return redirects
+    }
+)
+
+const resolveGrapherAndWordpressRedirect = async (url: Url): Promise<Url> => {
+    if (!url.pathname || !isCanonicalInternalUrl(url)) return url
+    const redirects = await getGrapherAndWordpressRedirectsMap()
+    const target = redirects.get(url.pathname)
+
+    if (!target) return url
+    const targetUrl = Url.fromURL(target)
+
+    return resolveGrapherAndWordpressRedirect(
+        // Pass query params through only if none present on the target (cf.
+        // netlify behaviour)
+        url.queryStr && !targetUrl.queryStr
+            ? targetUrl.setQueryParams(url.queryParams)
+            : targetUrl
```

Comment:
Is this definitely the Netlify behaviour? That if the target URL defines some query parameter, all incoming query parameters from the original URL are ignored? (Just asking, have no idea and haven't checked, but would really like to know.)

Reply:
@danielgavrilov yes, that's what I notice.

Considering the following redirect with no query params on the target (/public-spending/):
`/social-spending/ /public-spending/ 301!`

<img width="930" alt="Screenshot 2022-02-16 at 09 00 51" src="https://user-images.githubusercontent.com/13406362/154221245-ebfb9ae8-e952-4a7f-add0-95096657168e.png">

--> the query param on the source is passed through

Now with a query param on the target: 
`/grapher/child-mortality-by-country /grapher/child-mortality-around-the-world?country=KEN 301!`

<img width="927" alt="Screenshot 2022-02-16 at 08 56 22" src="https://user-images.githubusercontent.com/13406362/154221657-8ffe869f-c180-42c8-93e9-a37a26687a9e.png">

--> the query param on the source is neither passed through...

<img width="926" alt="Screenshot 2022-02-16 at 09 00 39" src="https://user-images.githubusercontent.com/13406362/154221766-9562e051-78a8-4c62-b57c-b7255e9f45b2.png">

--> ... nor merged

Reply:
Interesting, great to know, thanks!